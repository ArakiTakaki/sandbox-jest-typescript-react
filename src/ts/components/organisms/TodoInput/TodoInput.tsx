import React, { FormEventHandler, MouseEventHandler } from 'react';
import InputText from '../../atoms/InputText/InputText';
import Button from '../../atoms/Button/Button';
import styles from './style';

interface Props {
  onChangeInput: FormEventHandler;
  addTodoEvent: MouseEventHandler;
  inputTodo: string;
}
const TodoInput = (props: Props) => {
  return (
    <div className={styles.root}>
      <div className={styles.inputContent}>
        <InputText onChange={props.onChangeInput} value={props.inputTodo} borderBottom />
      </div>
      <div className={styles.button}>
        <Button onClick={props.addTodoEvent}>add</Button>
      </div>
    </div>
  );
};

export default TodoInput;
