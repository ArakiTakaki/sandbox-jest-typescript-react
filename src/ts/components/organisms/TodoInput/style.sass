.root
    display: flex
    align-items: center
    justify-content: space-between

.inputContent
    box-sizing: border-box
    width: 100%

.button
    margin-left: 2rem
