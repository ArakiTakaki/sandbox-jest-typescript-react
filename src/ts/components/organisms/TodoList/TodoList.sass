
.root
  margin: 1rem 0

.todo
  border-bottom: 1px solid #555
  padding: .5rem 0
  &:last-child
    border: 0