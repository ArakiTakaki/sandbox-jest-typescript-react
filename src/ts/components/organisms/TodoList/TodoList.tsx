import React from 'react';
import { Todo } from '~/store/model';
import TodoContent from '~/components/molecules/TodoContent/TodoContent';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

const styles = require('./TodoList.sass');
interface Props {
  todoList: Todo[];
  onDeleteTodo: React.MouseEventHandler;
  onBlur?: React.FocusEventHandler;
  onCompleted: React.MouseEventHandler;
}
interface State {}
class TodoList extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {};
    this.onChange = this.onChange.bind(this);
  }
  private onChange(event: React.FocusEvent<HTMLInputElement>) {
    const {
      value,
      dataset: { id },
    } = event.currentTarget;
    if (id == null) return;
    this.setState({ [id]: value });
  }

  render() {
    const { todoList, onDeleteTodo, onBlur, onCompleted } = this.props;
    return (
      <div className={styles.root}>
        <TransitionGroup component={'div'}>
          {todoList.map((todo: Todo, id: number) => (
            <CSSTransition key={todo.id} timeout={300} classNames={styles.list}>
              <div>
                <TodoContent
                  checked={todo.completed}
                  onDelete={onDeleteTodo}
                  onBlur={onBlur}
                  dataID={id}
                  onCompleted={onCompleted}
                >
                  {todo.name}
                </TodoContent>
              </div>
            </CSSTransition>
          ))}
        </TransitionGroup>
      </div>
    );
  }
}

export default TodoList;
