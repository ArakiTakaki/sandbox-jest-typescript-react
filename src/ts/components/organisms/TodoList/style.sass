.root
    margin: 1rem 0

.list
    display: flex
    flex-direction: column-reverse

.todo
    border-bottom: 1px solid #555
    padding: 0.5rem 0

    &:last-child
        border: 0

.list
    margin: 0

    &-enter
        overflow: hidden
        display: block
        opacity: 0.01
        max-height: 0vh

        &-active
            opacity: 1
            max-height: 10rem
            transition: 300ms ease-in
            transition-property: opacity, max-height

    &-exit
        overflow: hidden
        display: block
        opacity: 1
        max-height: 10rem

        &-active
            max-height: 0vh
            opacity: 0.01
            transition: 300ms ease-in
            transition-property: opacity, max-height
