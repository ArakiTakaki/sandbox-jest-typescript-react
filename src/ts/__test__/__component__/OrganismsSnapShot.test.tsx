/// <reference types="jest" />
import * as React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../../components/organisms/TodoList/TodoList';
import MOCK from './TodoModel';
import TodoInput from '../../components/organisms/TodoInput/TodoInput';
let mock = [...MOCK];

const indexProps = {
  todoList: mock,
  onDeleteTodo(idx: number): void {
    mock = mock.filter(item => item.id !== mock[idx].id);
    console.log(mock);
  },
  onCompleted(idx: number): void {
    console.log('accept onCompleted');
    console.log(idx);
  },
  onChangeName(idx: number, name: string): void {
    console.log(idx, name);
  },
};

test('TodoInput SnapShot', () => {
  const c = <TodoList {...indexProps} />;
  const component = renderer.create(c);
  expect(component.toJSON()).toMatchSnapshot();
});

const tProps = {
  addTodoEvent(name: string): void {},
};
test('TodoInput SnapShot', () => {
  const c = <TodoInput {...tProps} />;
  const component = renderer.create(c);
  expect(component.toJSON()).toMatchSnapshot();
});
