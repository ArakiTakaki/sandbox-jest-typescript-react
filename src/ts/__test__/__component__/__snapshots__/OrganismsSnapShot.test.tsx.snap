// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput SnapShot 1`] = `
<div
  className="root"
>
  <div
    className="list"
  >
    <div>
      <div
        className="root"
      >
        <div>
          <label>
            <input
              checked={false}
              className="checkbox"
              onChange={[Function]}
              type="checkbox"
              value={0}
            />
            <div
              className="icon"
            />
          </label>
        </div>
        <div
          className="inputText"
        >
          <div
            className="root"
          >
            <label>
              <input
                className="input"
                data-id={0}
                defaultValue="sample"
                onBlur={[Function]}
                type="text"
              />
              <div
                className="border"
              />
            </label>
          </div>
        </div>
        <div>
          <label>
            <input
              className="cross"
              onClick={[Function]}
              type="button"
              value={0}
            />
            <div
              className="icon"
            />
          </label>
        </div>
      </div>
    </div>
    <div>
      <div
        className="root"
      >
        <div>
          <label>
            <input
              checked={true}
              className="checkbox"
              onChange={[Function]}
              type="checkbox"
              value={1}
            />
            <div
              className="icon"
            />
          </label>
        </div>
        <div
          className="inputText"
        >
          <div
            className="root"
          >
            <label>
              <input
                className="input"
                data-id={1}
                defaultValue="test"
                onBlur={[Function]}
                type="text"
              />
              <div
                className="border"
              />
            </label>
          </div>
        </div>
        <div>
          <label>
            <input
              className="cross"
              onClick={[Function]}
              type="button"
              value={1}
            />
            <div
              className="icon"
            />
          </label>
        </div>
      </div>
    </div>
    <div>
      <div
        className="root"
      >
        <div>
          <label>
            <input
              checked={false}
              className="checkbox"
              onChange={[Function]}
              type="checkbox"
              value={2}
            />
            <div
              className="icon"
            />
          </label>
        </div>
        <div
          className="inputText"
        >
          <div
            className="root"
          >
            <label>
              <input
                className="input"
                data-id={2}
                defaultValue="component"
                onBlur={[Function]}
                type="text"
              />
              <div
                className="border"
              />
            </label>
          </div>
        </div>
        <div>
          <label>
            <input
              className="cross"
              onClick={[Function]}
              type="button"
              value={2}
            />
            <div
              className="icon"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoInput SnapShot 2`] = `
<div
  className="root"
>
  <div
    className="inputContent"
  >
    <div
      className="root"
    >
      <label>
        <input
          className="input bottom"
          data-test="organisms_input_text"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          className="border"
        />
      </label>
    </div>
  </div>
  <div
    className="button"
  >
    <div>
      <button
        className="my_button"
        data-test="organisms_submit_button"
        onClick={[Function]}
      >
        add
      </button>
    </div>
  </div>
</div>
`;
